geode-core/src/main/java/org/apache/geode/internal/statistics/AbstractExecutorGroup.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.internal.statistics;
+
+import java.lang.management.ManagementFactory;
+import java.lang.management.ThreadInfo;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Properties;
+
+import org.apache.logging.log4j.Logger;
+
+import org.apache.geode.distributed.ThreadMonitoring;
+import org.apache.geode.distributed.internal.DistributionConfigImpl;
+import org.apache.geode.internal.logging.LogService;
+
+public abstract class AbstractExecutorGroup {
+
+  private final ThreadMonitoring threadMonitoring;
+  private static final int THREAD_DUMP_DEPTH = 40;
+  private static final Logger logger = LogService.getLogger();
+  private long threadID;
+  private String grpName;
+  private short numIterationsStuck;
+  private long startTime;
+
+  private static final Properties nonDefault = new Properties();
+  private static final DistributionConfigImpl distributionConfigImpl =
+      new DistributionConfigImpl(nonDefault);
+
+  public AbstractExecutorGroup(ThreadMonitoring tMonitoring) {
+    this.startTime = System.currentTimeMillis();
+    this.numIterationsStuck = 0;
+    this.threadID = Thread.currentThread().getId();
+    this.threadMonitoring = tMonitoring;
+  }
+
+  public void handleExpiry(long stuckTime) {
+    this.incNumIterationsStuck();
+    logger.warn(handleLogMessage(stuckTime));
+    if (distributionConfigImpl.getThreadMonitorAutoEnabled()
+        && this.numIterationsStuck > distributionConfigImpl.getThreadMonitorAutoLimit())
+      handleInterrupt();
+  }
+
+  @SuppressWarnings("deprecation")
+  private void handleInterrupt() {
+    for (Thread t : Thread.getAllStackTraces().keySet()) {
+      if (t.getId() == this.threadID) {
+        logger.error("Thread: <{}> is stuck for <{}> iterations - Executing Interrupt\n",
+            this.threadID, this.numIterationsStuck);
+
+        t.stop();
```

Comment:
Isn't `Thread.stop()` is deprecated as being inherently unsafe?

Reply:
indeed this method is deprecated
OOB this function is disabled -
if (distributionConfigImpl.getThreadMonitorAutoEnabled()
        && this.numIterationsStuck > distributionConfigImpl.getThreadMonitorAutoLimit())
      handleInterrupt();

although in all of our runs when pool thread was killed it continued to work on next task when it came(with same ID and name) and we saw no real issue. also we need to take into consideration that if the thread is not stopped it can remain stuck until entire JVM is bounced (as we faced)